### PR TITLE
Clean up ssh tunnels during exit.

### LINKED ISSUE #10752: SSH tunnels left open after sending SIGINT to `minikube tunnel`
**Steps to reproduce the issue:** 
Minikube running with docker driver on macOS Catalina

1. `minikube tunnel`
2. On another terminal: `kill -INT <PID_OF_THE_PREVIOUS_COMMAND>`
3. `ps aux | grep minikube`

**Full output of failed command:** 
The `ps` command will show multiple ssh tunnels still open, outliving the parent process.

**Details**
In **pkg/minikube/tunnel/kic/ssh_tunnel.go** the interrupt signal is caught and it stops the LoadBalancerEmulator. However, it assumes that the interrupt will also propagate to the children ssh tunnels, which are running in a goroutine inside `startConnections`.
The interrupt, however, kills the goroutine and the blocking call which waits for the child process to finish. This leaves the ssh tunnels dangling.
On the other hand, if `CTRL + C` is sent on the same terminal as `minikube tunnel` and the clean-up routine included killing all ssh tunnels, the OS would error with "process already finished".

**Related issues**
https://github.com/kubernetes/minikube/issues/8511
https://github.com/kubernetes/minikube/issues/3647

### PR DESCRIPTION
**Pull request by vishjain**
fixes #10752 

Before PR: 
  PID TTY           TIME CMD
 1417 ttys000    0:00.14 /bin/bash --rcfile /Applications/GoLand.app/Contents/plugins/terminal/jediterm-bash.in -i
 1696 ttys001    0:00.07 login -pf vishal
 1698 ttys001    0:00.12 -bash
 7548 ttys001    0:00.25 minikube tunnel
 7565 ttys001    0:00.25 minikube tunnel

After PR (Deleting is successful with no stray tunnels) : 
  PID TTY           TIME CMD
 1417 ttys000    0:00.14 /bin/bash --rcfile /Applications/GoLand.app/Contents/plugins/terminal/jediterm-bash.in -i
 1696 ttys001    0:00.07 login -pf vishal
 1698 ttys001    0:00.12 -bash

### CODE REVIEW

**Review by medyagh — CHANGES_REQUESTED** [review]
could u plz post before/after this PR in the PR Description ?

**Comment by vishjain** [comment]
> could u plz post before/after this PR in the PR Description ?

Done.

**Comment by vishjain** [comment]
@medyagh Can you take a glance at this, Thanks!

File: pkg/minikube/tunnel/kic/ssh_conn.go [hunk 1]
```diff
@@ -142,14 +145,21 @@ func (c *sshConn) startAndWait() error {
...
+	c.activeConn = false
+
 	return nil
 }
 
 func (c *sshConn) stop() error {
-	out.Step(style.Stopping, "Stopping tunnel for service {{.service}}.", out.V{"service": c.service})
-
-	return c.cmd.Process.Kill()
+	if c.activeConn {
```
Comment (ilya-zuyev) [inline pkg/minikube/tunnel/kic/ssh_conn.go:159]: can we set `c.activeConn` to `false` here? If somehow we call `c.stop()` twice, we'll try to kill already stopped process.

**Comment by sharifelgamal** [comment]
@vishjain This looks good, could you just address the one comment above, then we can go ahead and run the full test suite.

**Comment by vishjain** [comment]
> @vishjain This looks good, could you just address the one comment above, then we can go ahead and run the full test suite.

I've made the necessary changes here.

**Pull request merged**